### ISSUE TITLE
Compression lossy or not?

### ISSUE THREAD

**Issue by jancborchardt** [stefansl/image-shrinker#42]
It seems that the compression is lossy? This is a PNG logo I compressed the other day.

Original:
![image-shrinker original](https://user-images.githubusercontent.com/925062/40744950-68d89232-6456-11e8-9eab-612d806b87b6.png)

Shrunk file, note the staircase effect:
![image-shrinker shrunk hard edge](https://user-images.githubusercontent.com/925062/40744949-68b418b2-6456-11e8-91bd-935d3f5fbc4a.png)

(In Nextcloud for example we use [Trimage for compression of JPG/PNG](https://trimage.org/) and [scour for compression of SVG](https://github.com/nextcloud/server/blob/master/build/image-optimization.sh#L10), which are both lossless and have not had issues. Lossy compression like SVGO and others have always led to issues, so we avoid them.)

Especially for SVG it needs to be not lossy since they are often the source files. If some of the methods used in Image Shrinker are lossy and others not, maybe a switch to "Compress even more! (smaller files but lossy compression)" or the like is best?

**Comment by jancborchardt** [comment]
Ok, so according to the projects:
- [pngquant](https://pngquant.org/) is lossy (also see example above)
- [mozjpeg](https://github.com/mozilla/mozjpeg#mozilla-jpeg-encoder-project) is lossless
- [svgo](https://github.com/svg/svgo/) seems lossless if used with [the right attributes](https://github.com/svg/svgo/issues/646), cause it also seems to have lossy ones. (As said we had bad experience using it [in Nextcloud](https://github.com/nextcloud/server/issues/4076).)

**Comment by stefansl** [comment]
Good point, I love mozjpeg - for me it's the best,  jpg compressor - it's quite fast and the results are great. 
But I will check out scour and optipng, pngcrush, advpng. Those png compressors are used by Trimage
